1. Когда используется критерий Стьюдента, а когда Z –критерий?

Если неизвестна сигма генеральной совокупности и соблюдаются остальные условия применимости параметрических тестов, то следует использовать критерий Стьюдента.
Если сигма генеральной совокупности  известна и соблюдается нормальность распределения генеральной совокупности, то предпочтение отдают Z критерию.

2. Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.

mu = 17 (средний диаметр шариков генеральной совокупности)
alpha = 0.05 = 5% (односторонний критерий)
n = 100 (шариков)
X = 17.5 (средний диаметр шариков выборки)
D(x) = 4 (дисперсия в кв.мм)
sigma = sqrt(D(x)) (среднее квадратичное отклонение, корень из дисперсии) = 2

Z=((X-μ)/σ/√n
Z = (17.5-17)/2 = 2.5

По таблице, для 0.95 Z = 1.695
Так как Z(фактическое) > Z(табличное) то принимаем альтернативную гипотезу. 

3. Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г.

Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190.

Известно, что их веса распределены нормально.

Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

mu = 200 (средний вес одной пачки)
n = 10 (выборка из 10 пачек)
нормальное распределение
доверительная вероятность = 99%, значит, что alpha = 1%
H_0 (нулевая гипотеза)  M = 200
H_1 (альтернативная) M != 200
неизвестна сигма генеральной совокупности, значит будет использоваться t-критерий
так как необходимо произвести двусторонний тест, то t будет рассчитано, как t(alpha / 2)

In [6]:
import numpy as np
import scipy.stats as stats
x = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
mean_x = np.mean(x)
std_x = np.std(x,ddof=1)
std_x

4.453463071962462

t_н=(X ̅-μ)/(σ_н/√n)

In [8]:
t_h = (mean_x - 200)/(std_x/np.sqrt(10))
t_h

-1.0651074037450896

Так как выборка у нас из 10 пачек, значит смотрим степень свободы для 9 и alpha/2=0.005. t- распределение = 3,250.
Так как -3.250<-1.065<3.250, мы можем утверждать, что утверждения продавца верны.

4. Есть ли статистически значимые различия в росте дочерей?

Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165

Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [12]:
import numpy as np 
import scipy.stats as stats

x = np.array([172, 177, 158, 170, 178, 175, 164, 160, 169, 165])
y = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])
stats.ttest_ind(x,y)

Ttest_indResult(statistic=0.24400587215969016, pvalue=0.8101467126543233)

Т.к. значение pvalue=0.810, слишком высокое, значение a=0.005, мы можем сделать вывод, что статистически значимых различий в росте дочерей нет.